CRADLE/CorrectBiasStored/correctBias.py

```diff
@@ -18,6 +18,89 @@
 RC_PERCENTILE = [0, 20, 40, 60, 80, 90, 92, 94, 96, 98, 99, 100]
 
 
+def divideWork(regionSet: list[tuple[str, int, int]], totalBaseCount: int, numProcesses: int) -> list[list[tuple[str, int, int]]]:
+	""" Break a list of regions into _numProcesses_ separate lists of roughly equal length (as measured by number of base pairs
+	covered by the list)
+	"""
+	idealWorkSize = math.ceil(totalBaseCount / numProcesses)
+
+	# "Priming" currentJobList with the first region helps us avoid some edge cases (e.g., the first region is much bigger than
+	# idealWorkSize) where the first list in allJobs would be empty.
+	initChromo, initStart, initEnd = regionSet[0]
+	currentJobSize = initEnd - initStart
+	currentJobList = [(initChromo, initStart, initEnd)]
+	allJobs = []
+	for chromo, start, end in regionSet[1:]:
+		regionLength = end - start
+		newJobSize = currentJobSize + regionLength
+
+		# We want len(allJobs) to equal numProcesses so if len(allJobs) == numProcesses - 1
+		# then we are currently filling in the very last work set and all the rest of the
+		# regions should be added to currentJobList. Once the rest of the regions are added
+		# to currentJobList the for loop will end and currentJobList will be appended to allJobs.
+		# This will make len(allJobs) == numProcesses, like we want.
+		if newJobSize < idealWorkSize or len(allJobs) == numProcesses - 1:
+			currentJobSize = newJobSize
+			currentJobList.append((chromo, start, end))
+		elif newJobSize - idealWorkSize < (regionLength / 5): # It's just a little too big, but that's okay.
```

Comment:
What do you think of choosing a reasonable number instead of `regionLen/5`?

Reply:
I have no idea what a better number here would be. If you have a preference, I'm open to changing it.

Reply:
or why don't we use `idealWorkSize/5` instead of 'regionLength / 5'?

Reply:
`idealWorkSize/5` could be a pretty big number numProcesses is like 2. it would be the `totalBaseCount / 10` I don't think it's a good idea for our "wiggle-room" to depend on how many processes we're going to run.

Reply:
okay. Then let's use the way it is now.